src/test/java/ch/uzh/ifi/hase/soprafs22/service/UserServiceTest.java

```diff
@@ -165,9 +263,413 @@ void updateUser_Conflict(){ //Test what happens when user tries to change userna
+        userService.setMatch(testMatch);
+        assertTrue(userService.doesMatchExist(testUser, otherUser));
+        assertTrue(userService.doesMatchExist(otherUser, testUser));
+
+        //case multiple matches exist
+        User thirdUser = new User();
+        thirdUser.setId(13L);
+        Match otherMatch = userService.createMatch(testUser, thirdUser);
+        otherMatch.setMatchId(23L);
+
+        Mockito.when(matchRepository.findByMatchId(23L)).thenReturn(otherMatch);
+        userService.setMatch(otherMatch);
+        assertTrue(userService.doesMatchExist(testUser, thirdUser));
+        assertTrue(userService.doesMatchExist(thirdUser,testUser));
+    }
+
+    @Test
+    void doesMatchExist_false(){
+        // given
+        User testUser2 = new User();
+        testUser2.setId(3L);
+        testUser2.setName("testName");
+        testUser2.setUsername("testUsername3");
+        testUser2.setPassword("1234");
+        testUser2.setGender(Gender.OTHER);
+
+        Match testMatch = userService.createMatch(otherUser, testUser2);
+        testMatch.setMatchId(2L);
+
+        Mockito.when(matchRepository.findByMatchId(2L)).thenReturn(testMatch);
+        userService.setMatch(testMatch);
+        assertFalse(userService.doesMatchExist(testUser, otherUser));
+        assertFalse(userService.doesMatchExist(otherUser, testUser));
+    }
+
+    @Test
+    void checkGeneralAccess_granted(){
+      testUser.setToken("token");
+      Mockito.when(userRepository.findByToken(testUser.getToken())).thenReturn(testUser);
+      userService.checkGeneralAccess(testUser.getToken());
+      Mockito.verify(userRepository, Mockito.times(1)).findByToken(Mockito.any());
+    }
+
+    @Test
+    void checkGeneralAccess_denied(){
+        testUser.setToken("token");
+        Mockito.when(userRepository.findByToken(testUser.getToken())).thenReturn(null);
+        ResponseStatusException e = assertThrows(ResponseStatusException.class, ()-> userService.checkGeneralAccess("token"));
+        assertEquals(HttpStatus.UNAUTHORIZED, e.getStatus());
+    }
+
+    @Test
+    void getCurrentBlackCards_emptyListReturned(){
+      Mockito.when(userRepository.findById(1L)).thenReturn(testUser);
+      assertEquals(Collections.emptyList(), userService.getCurrentBlackCards(testUser.getId()));
+      testUser.setUserBlackCards(null);
+      assertEquals(Collections.emptyList(), userService.getCurrentBlackCards(testUser.getId()));
+      testUser.setUserBlackCards(new UserBlackCards());
+      assertEquals(Collections.emptyList(), userService.getCurrentBlackCards(testUser.getId()));
+      List<BlackCard> cards = new ArrayList<>();
+      UserBlackCards ubc = new UserBlackCards();
+      ubc.setBlackCards(cards);
+      assertEquals(Collections.emptyList(), userService.getCurrentBlackCards(testUser.getId()));
+      testUser.setUserBlackCards(userBlackCards);
+      // ToDo: for 100% line coverage: test with Powermock to access private final Date of userBlackCards
```

Comment:
I dont think we need 100%

Reply:
Nor do I.. but I think this is a rather important feature that is why I have put it up for debate..

Reply:
makes sense